Source/WebCore/editing/VisibleSelection.cpp

```diff
@@ -683,8 +683,8 @@ Node* VisibleSelection::nonBoundaryShadowTreeRootNode() const
 
 bool VisibleSelection::isInPasswordField() const
 {
-    RefPtr textControl = enclosingTextFormControl(start());
-    return is<HTMLInputElement>(textControl) && downcast<HTMLInputElement>(*textControl).isPasswordField();
+    RefPtr textControl = dynamicDowncast<HTMLInputElement>(enclosingTextFormControl(start()));
```

Comment:
I didn't look into `isPasswordField()` but I doubt we need a smart pointer here.

Reply:
This was also added as part of 389241b67619b49fab63ca2d9abafb5c2a69358a. `isPasswordField()` is a somewhat trivial getter (although `HTMLInputElement` does delegate it to `InputType`). Happy to remove if you confirm.

Maybe that suggests that commit needs another review given the lessons learned in the past 4 months?